src/index.js

```diff
@@ -1,65 +1,172 @@
+        context.fail("Exception: " + e);
+    }
 };
 
-// Extend AlexaSkill
-ReverseGeek.prototype = Object.create(AlexaSkill.prototype);
-ReverseGeek.prototype.constructor = ReverseGeek;
-
-ReverseGeek.prototype.eventHandlers.onLaunch = function (launchRequest, session, response) {
+function onSessionStarted(sessionStartedRequest, session) {
+    console.log("onSessionStarted requestId=" + sessionStartedRequest.requestId +
+        ", sessionId=" + session.sessionId);
+}
+
+function onLaunch(launchRequest, session, callback) {
+    console.log("onLaunch requestId=" + launchRequest.requestId +
+        ", sessionId=" + session.sessionId);
+
+     getWelcomeResponse(callback);
+}
+
+function onIntent(intentRequest, session, callback) {
+    console.log("onIntent requestId=" + intentRequest.requestId +
+        ", sessionId=" + session.sessionId);
+
+    var intent = intentRequest.intent,
+        intentName = intentRequest.intent.name;
+
+    if ("ReverseWord" === intentName) {
+        setWordInSession(intent, session, callback);
+    } else if ("AMAZON.HelpIntent" === intentName) {
+        getWelcomeResponse(callback);
+    } else if ("AMAZON.StopIntent" === intentName || "AMAZON.CancelIntent" === intentName) {
+        handleSessionEndRequest(callback);
+    } else {
+        throw "Invalid intent";
+    }
+}
+
+function onSessionEnded(sessionEndedRequest, session) {
+    console.log("onSessionEnded requestId=" + sessionEndedRequest.requestId +
+        ", sessionId=" + session.sessionId);
+
+}
+
+function getWelcomeResponse(callback) {
+    // If we wanted to initialize the session to have some attributes we could add those here.
+    var sessionAttributes = {};
+    var cardTitle = "Welcome";
+    var speechOutput = "Hi there! I'm reverse geek. " +
+        "What word do you want me to reverse?";
+    // If the user either does not reply to the welcome message or says something that is not
+    // understood, they will be prompted again with this text.
+    var repromptText = "What word do you want to reverse? You can say \"reverse carwheel\".";
+    var shouldEndSession = false;
+
+    callback(sessionAttributes,
+        buildSpeechletResponse(cardTitle, speechOutput, repromptText, shouldEndSession));
+}
+
+function handleSessionEndRequest(callback) {
+    var cardTitle = "Session Ended";
+    var speechOutput = "Thank you for trying reverse geek. Have a nice day!";
+    // Setting this to true ends the session and exits the skill.
+    var shouldEndSession = true;
+
+    callback({}, buildSpeechletResponse(cardTitle, speechOutput, null, shouldEndSession));
+}
+
+function setWordInSession(intent, session, callback) {
+    var cardTitle = intent.name;
+    var wordSlot = intent.slots.Word;
+    var repromptText = "";
+    var sessionAttributes = {};
+    var shouldEndSession = true;
+    var speechOutput = "";
+
+    if (wordSlot.value) {
+        var word = wordSlot.value;
+        // sessionAttributes = createwordAttributes(word);
```

Comment:
I commented this guy out because I didn't see the purpose, but left it so we could discuss it. I checked out ```createwordAttributes(word)```, which just returns the slot value in JSON format.